Subsurface/Node/Common/Peripherals/src/SPIInterface.cpp

```diff
@@ -0,0 +1,121 @@
+#include "Peripherals/inc/SPIInterface.h"
+
+SPIInterface::SPIInterface(SPI_HandleTypeDef *hspi) : m_hspi(hspi){}
+
+HAL_StatusTypeDef SPIInterface::Transmit(uint8_t* data, uint16_t size,
+                                         uint32_t timeout)
+{
+    return HAL_SPI_Transmit(m_hspi, data, size, timeout);
+}
+
+HAL_StatusTypeDef SPIInterface::Receive(uint16_t size, uint32_t timeout)
+{
+    return HAL_SPI_Receive(m_hspi, m_dataReceived, size, timeout);
+}
+
+HAL_StatusTypeDef SPIInterface::TransmitReceive(uint8_t* dataOut,
+                                                uint16_t size, uint32_t timeout)
```

Comment:
I think the formatting for this would look better like
```c
HAL_StatusTypeDef SPIInterface::TransmitReceive(uint8_t* dataOut,
                                                uint16_t size,
                                                uint32_t timeout)
```

Reply:
Formatted to 120 char limit.